streams/src/main/java/org/apache/kafka/streams/state/WindowStore.java

```diff
@@ -27,20 +27,25 @@
 public interface WindowStore<K, V> extends StateStore, ReadOnlyWindowStore<K, V> {
 
     /**
-     * Put a key-value pair with the current record time as the timestamp
-     * into the corresponding window
+     * Use the current record timestamp as the {@code windowStartTimestamp} and
+     * delegate to {@link WindowStore#put(Object, Object, long)}.
+     *
      * @param key The key to associate the value to
      * @param value The value to update, it can be null;
      *              if the serialized bytes are also null it is interpreted as deletes
      * @throws NullPointerException If null is used for key.
+     *
+     * @deprecated since 2.1. Use {@link WindowStore#put(Object, Object, long)} instead.
      */
+    @Deprecated
```

Comment:
`WindowStore` is public API -- we need a KIP if we want to deprecate something. Thus, this is not a `MINOR` PR.

Reply:
Roger that.